### PR TITLE
Pull Request (Restructure Action & Multiple Selections)

### PR DESCRIPTION
**Pull request by DawidSA2017**

### CODE REVIEW

**Comment by Kogoro** [comment]
- Two `Export As` entries when clicking on feature diagram editor with no selection
- `Show Hidden Features` and `Show Collapsed Constraints` seem to be misplaced (Maybe better under `Show Legend`)
- Collapse Siblings could also work when multiple features are selected
- A feature without children is automatically collapsed (checked as menu entry), but should not be saved (please prove)

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,4 @@
 /plugins/de.ovgu.featureide.examples/featureide_examples/StringMatcher-FH-JML/src
 /tests/de.ovgu.featureide.fm.core-test/.featureide/
 /tests/de.ovgu.featureide.fm.core-test/feature_model_tmp**
+
```
Comment (Kogoro) [inline .gitignore]: This file seems to be untouched with no reason to commit.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MultipleSelectionAction.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+				tempFeature.addListener(this);
+			}
+			updateProperties();
+		} else {
+			featureArray = null;
+			setEnabled(false);
+		}
+	}
+	
+	protected abstract void updateProperties();
```
Comment (Kogoro) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MultipleSelectionAction.java]: There should be some kind of method header here

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/MandatoryFeatureOperation.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import static de.ovgu.featureide.fm.core.localization.StringTable.MANDATORY_OPERATION;
+
+import de.ovgu.featureide.fm.core.base.IFeature;
+import de.ovgu.featureide.fm.core.base.IFeatureModel;
+import de.ovgu.featureide.fm.core.base.event.FeatureIDEEvent;
+import de.ovgu.featureide.fm.core.base.event.FeatureIDEEvent.EventType;
+
+/**
+ * TODO description
```
Comment (Kogoro) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/MandatoryFeatureOperation.java]: There should be a description

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CollapseAction.java [hunk 4]
```diff
@@ -66,41 +79,150 @@ public void selectionChanged(SelectionChangedEvent event) {
 		}
 	};
 
-	public CollapseAction(Object viewer, IGraphicalFeatureModel graphicalFeatureModel) {
+	public CollapseAction(Object viewer, IGraphicalFeatureModel graphicalFeatureModel, IFeatureModel featureModel) {
```
[the remark this answers is not in the document]
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/CollapseAction.java]: fixed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MultipleSelectionAction.java (hunk 2 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/MultipleSelectionAction.java]: fixed

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/MandatoryFeatureOperation.java (hunk 3 above)
Comment (PaulWestphal) [reply plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/MandatoryFeatureOperation.java]: fixed

**Comment by PaulWestphal** [comment]
>Two Export As entries when clicking on feature diagram editor with no selection

fixed

>Show Hidden Features and Show Collapsed Constraints seem to be misplaced (Maybe better under Show Legend) (Also with no selection)

We discussed this with @tthuem and decided that having "expand/collapse all" grouped with "show hidden features" and "show hidden constraints" makes sense, since it also changes which features you see.

>A feature without children is automatically collapsed (checked as menu entry), but should not be saved (please prove)

The checkbox next to the collapse action is now always unchecked when the action is disabled.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java [hunk 5]
```diff
@@ -728,8 +728,6 @@ else if ((hiddenAction.isEnabled()) && !connectionSelected) {
 				extension.extendContextMenu(menu, this);
 			}
 		}
-		menu.add(new Separator(IWorkbenchActionConstants.MB_ADDITIONS));
-		menu.add(this.exportFeatureModelAction);
```
Comment (Kogoro) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/FeatureDiagramEditor.java]: This change removes the Export As entry from almost all context menus. This is not intended.

**Comment by Kogoro** [comment]
Now there are changes to an automotive example. Also, the export as entry issue is still present.
Could you also comment on the idea with collapsing siblings with multiple features?

**Comment by PaulWestphal** [comment]
>Now there are changes to an automotive example. 
This branch is a few commits behind, the changes to the automotive example are part of a commit by @tthuem (https://github.com/FeatureIDE/FeatureIDE/commit/803fea1c74109d49663dfc1c70ba0c3f4bcd33fb).

>Also, the export as entry issue is still present.
It's not an issue, it's a feature. It's misleading if you select one or more features because you always export the entire model.

>Could you also comment on the idea with collapsing siblings with multiple features?
We discussed this with @tthuem a week ago and decided that it was not necessary.

**Pull request merged**